libyara/modules/lnk/lnk.c

```diff
@@ -0,0 +1,2339 @@
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_integer("code_page");
+  end_struct("console_fe_data");
+
+  declare_integer("has_darwin_data");
+
+  begin_struct("darwin_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_string("darwin_data_ansi");
+    declare_string("darwin_data_unicode");
+  end_struct("darwin_data");
+
+  declare_integer("has_environment_variable_data");
+
+  begin_struct("environment_variable_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_string("target_ansi");
+    declare_string("target_unicode");
+  end_struct("environment_variable_data");
+
+  declare_integer("has_icon_environment_data");
+
+  begin_struct("icon_environment_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_string("target_ansi");
+    declare_string("target_unicode");
+  end_struct("icon_environment_data");
+
+  declare_integer("has_known_folder_data");
+
+  begin_struct("known_folder_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_integer("offset");
+    declare_integer_array("known_folder_id");
+  end_struct("known_folder_data");
+
+  declare_integer("has_property_store_data");
+
+  begin_struct("property_store_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+  end_struct("property_store_data");
+
+  declare_integer("has_shim_data");
+
+  begin_struct("shim_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_string("layer_name");
+  end_struct("shim_data");
+
+  declare_integer("has_special_folder_data");
+
+  begin_struct("special_folder_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_integer("special_folder_id");
+    declare_integer("offset");
+  end_struct("special_folder_data");
+
+  declare_integer("has_tracker_data");
+
+  begin_struct("tracker_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_string("machine_id");
+    declare_string("droid_volume_identifier");
+    declare_string("droid_file_identifier");
+    declare_string("droid_birth_volume_identifier");
+    declare_string("droid_birth_file_identifier");
+  end_struct("tracker_data");
+
+  declare_integer("has_vista_and_above_id_list_data");
+
+  begin_struct("vista_and_above_id_list_data");
+    declare_integer("block_size");
+    declare_integer("block_signature");
+    declare_integer("number_of_item_ids");
+
+    begin_struct_array("item_id_list");
+      declare_integer("size");
+      declare_string("data");
+    end_struct_array("item_id_list");
+  end_struct("vista_and_above_id_list_data");
+
+  declare_integer("has_overlay");
+  declare_integer("overlay_offset");
+
+  declare_integer("is_malformed");
+end_declarations
+
+unsigned int parse_id_list(
```

Comment:
Better use `uint32_t` or `uint64_t`. The size of `unsigned int` depends on the platform and this may cause issues while compiling on 32-bits systems. The same applies to the rest of the functions.

Reply:
Thanks, I've updated this throughout!